File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/typescript/end-users/validateAccessToken.ts
// Usage: CDP_ACCESS_TOKEN=... pnpm tsx end-users/validateAccessToken.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();

const accessToken = process.env.CDP_ACCESS_TOKEN;
if (!accessToken) {
    throw new Error("CDP_ACCESS_TOKEN is not set");
}

try {
    const endUser = await cdp.endUser.validateAccessToken({
        accessToken,
    });
    console.log("Access token validated: ", endUser);
} catch (error) {
    console.error("Error validating access token: ", (error as { errorMessage: string }).errorMessage);
}

File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/python/end_user/validate_access_token.py
# Usage: CDP_ACCESS_TOKEN=... uv run python end_user/validate_access_token.py

import asyncio
from cdp import CdpClient
import os
from dotenv import load_dotenv

load_dotenv()

access_token = os.getenv("CDP_ACCESS_TOKEN")

if not access_token:
    raise ValueError("CDP_ACCESS_TOKEN must be set")


async def main():
    async with CdpClient() as cdp:
        try:
            end_user = await cdp.end_user.validate_access_token(
                access_token=access_token,
            )
            print(end_user)
        except Exception as e:
            # Access token is invalid or expired
            raise e


asyncio.run(main()) 